test/e2e/main_test.go

```diff
@@ -108,24 +111,97 @@ func TestNodeStartup(t *testing.T) {
 	pod := corev1.Pod{}
 	testutil.ExpectNoError(kubernetes.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: spec.Name, Namespace: tconst.Namespace}, &ispn))
 	testutil.ExpectNoError(kubernetes.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: spec.Name + "-0", Namespace: tconst.Namespace}, &pod))
-	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] ||
-		pod.Labels["operator-pod-label"] != "operator-pod-value" ||
-		ispn.Labels["operator-pod-label"] != "operator-pod-value" {
-		panic("Labels haven't been propagated to pods")
+
+	// Checking labels propagation to pods
+	// from Infinispan CR to pods
+	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] {
+		panic("Infinispan CR labels haven't been propagated to pods")
 	}
 
+	// from operator environment
+	if tconst.RunLocalOperator == "TRUE" {
+		// running locally, labels are hardcoded and set by the testsuite
+		if pod.Labels["operator-pod-label"] != "operator-pod-value" ||
+			ispn.Labels["operator-pod-label"] != "operator-pod-value" {
+			panic("Infinispan CR labels haven't been propagated to pods")
+		}
+	} else {
+		// Get the operator namespace from the env if it's different
+		// from the testsuite one
+		operatorNS := tconst.OperatorNamespace
+		if operatorNS == "" {
+			operatorNS = spec.Namespace
+		}
+		// operator deployed on cluster, labels are set by the deployment
+		if !areOperatorLabelsPropagated(spec.Namespace, ispnv1.OperatorPodTargetLabelsEnvVarName, pod.Labels) {
```

Comment:
Shouldn't be `operatorNS` propagated here instead of `spec.Namespace`?

Reply:
indeed, thanks @Crumby